demos/interactive_face_detection_demo/detectors.cpp

```diff
@@ -613,10 +613,15 @@ std::string EmotionsDetection::operator[] (int idx) const {
     }
 
     auto emotionsValues = emotionsBlob->buffer().as<float *>();
-    auto outputIdxPos = emotionsValues + idx;
+    auto outputIdxPos = emotionsValues + idx*numOfChannels;
 
     /* identify an index of the most probable emotion in output array
            for idx image to return appropriate emotion name */
+    std::string emotionsValuesRepr = "";
+    for (auto emValPtr=outputIdxPos; emValPtr<outputIdxPos + emotionsVecSize; emValPtr++){
+        emotionsValuesRepr +=  "(" + emotionsVec[emValPtr-outputIdxPos] + ":" + std::to_string(*emValPtr) + ");";
+    }
+    slog::info << "Face emotions:" << emotionsValuesRepr << slog::endl;
```

Comment:
Also, I would ask to remove this source code because tracking of resulting emotions is already supported by FLAGS_r option enabled (it means raw output enabled). Please double check. If this flag is enabled all detection/recognition results are printed to the console.

Reply:
It looks for me that only the max score emotion is logged with -r flag. There are examples of demo execution in the files attached to the PR ([face_emotions_after_fix.txt](https://github.com/opencv/open_model_zoo/files/2856588/face_emotions_after_fix.txt)) with -r flag, I haven't noticed the same logged values. Am I missing something?
If these logs are still unnecessary for any reason, I can remove them;  I have added them just because I could not find this info in logs myself.
Please specify once again if it is better to remove them anyway, thanks.

Reply:
It should be placed under _r flag at least because in case of long video input passed we will get overwritten console output. Moreover, what's the reason to track all of these values in real time? which benefit it gives for a user? I think the necessary value is only max prob emotion which we show currently. Other values are smaller and not so interesting for a user, isn't it? I still think it should be removed and you can add such debug information to own application any time when needed. But official version doesn't need it I think... There are many places where we could use detailed tracing but we avoid it to not overload samples/demos by such long debug info.

Reply:
Sounds very reasonable, I have removed this part.